File: translate-bubble/src/main/java/com/fortysevendeg/translace_bubble/receivers/BootReceiver.java
package com.fortysevendeg.translace_bubble.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import com.fortysevendeg.translace_bubble.ui.services.BubbleUIService;

public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        BubbleUIService.launchIfIsNecessary(context);
    }

}

File: src/main/scala/com/fortysevendeg/translatebubble/receivers/BootReceiver.scala
package com.fortysevendeg.translatebubble.receivers

import android.content.{Intent, Context, BroadcastReceiver}
import com.fortysevendeg.translatebubble.ui.bubbleservice.BubbleService
import macroid.AppContext

class BootReceiver extends BroadcastReceiver {
  override def onReceive(context: Context, intent: Intent): Unit = {
    implicit lazy val appCtx = AppContext(context.getApplicationContext)
    BubbleService.launchIfIsNecessary
  }

}
